services/conversation/v1-exp.js

```diff
@@ -0,0 +1,143 @@
+/**
+ * Copyright 2016 IBM Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+module.exports = function (RED) {
+  var cfenv = require('cfenv'), watson = require('watson-developer-cloud'), service = null,
+    sUsername = null, sPassword = null;
+  
+  service = cfenv.getAppEnv().getServiceCreds(/conversation/i);
+
+  if (service) {
+    sUsername = service.username;
+    sPassword = service.password;
+  }
+
+  RED.httpAdmin.get('/watson-conversation/vcap', function (req, res) {
+    res.json(service ? {bound_service: true} : null);
+  });
+ 
+  
+  function verifyPayload(node, msg) {
+    if (!msg.payload) {
+      this.status({fill:'red', shape:'ring', text:'missing payload'}); 
+      node.error('Missing property: msg.payload', msg);
+      return false;
+    }
+    return true;
+  }
+
+  function verifyInputs(node, msg, config) {
+    // workspaceid can be either configured in the node, 
+    // or sent into msg.params.workspace_id
+    if (!config.workspaceid || '' !== config.workspaceid) {
+      node.workspaceid = config.workspaceid;
+      return true;
+    } else if (!msg.params || '' !== msg.params.workspace_id) {
+      node.workspaceid = msg.params.workspace_id;
+      return true;
+    }
+    node.error('Missing workspace_id. check node documentation.',msg);
+    return false;
+  }
+
+  function verifyServiceCredentials(node, msg) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Function 'verifyServiceCredentials' has a complexity of 5.](https://www.codacy.com/app/node-red-nodes-watson/node-red-node-watson/file/2777516067/issues/source?bid=3225115&fileBranchId=3364374#l56)